packages/hoprd/crates/hoprd-hoprd/src/main.rs

```diff
@@ -0,0 +1,161 @@
+use utils_log::{info, warn};
+use utils_types::traits::{PeerIdLike, ToHex};
+use crate::cli::CliArgs;
+
+
+const ONBOARDING_INFORMATION_INTERVAL: std::time::Duration = std::time::Duration::from_secs(30);
+    // // Metrics
+    // TODO: introduce RLP
+    // const metric_latency = create_histogram_with_buckets(
+    //   'hoprd_histogram_message_latency_ms',
+    //   'Histogram of measured received message latencies',
+    //   new Float64Array([10.0, 25.0, 50.0, 100.0, 250.0, 500.0, 1000.0, 2500.0, 5000.0, 10000.0, 20000.0])
+    // )
+
+
+#[async_std::main]
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    env_logger::init();
+
+    info!("This is HOPRd {}", hopr_lib::constants::APP_VERSION);
+    let args = <CliArgs as clap::Parser>::parse();
+
+    // TOOD: add proper signal handling
+    // The signal handling should produce the crossbeam-channel and notify all background loops to terminate gracefully
+    // https://rust-cli.github.io/book/in-depth/signals.html
+
+    // TODO: remove the dry run
+    if ! args.dry_run {
+        if std::env::var("DAPPNODE").map(|v| v.to_lowercase() == "true").unwrap_or(false) {
+            info!("The HOPRd node appears to run on DappNode");
+        }
+
+        let cfg = config::HoprdConfig::from_cli_args(args, false)?;
+        info!("Node configuration: {}", cfg.as_redacted_string()?);
+
+        // Find or create an identity
+        let identity_opts = IdentityOptions {
+            initialize: cfg.hopr.db.initialize,
+            id_path: cfg.identity.file.clone(),
+            password: cfg.identity.password.clone(),
+            use_weak_crypto: Some(cfg.test.use_weak_crypto),
+            private_key: cfg.identity.private_key.clone().and_then(|v| cli::parse_private_key(&v).ok()),
+        };
+
+        let hopr_keys = HoprKeys::init(identity_opts)?;
+
+        info!("This node has packet key '{}' and uses a blockchain address '{}'",
+            core_transport::Keypair::public(&hopr_keys.packet_key).to_peerid_str(),
+            core_transport::Keypair::public(&hopr_keys.chain_key).to_hex()
+        );
+
+        // TODO: the following check can be removed once [PR](https://github.com/hoprnet/hoprnet/pull/5665) is merged
+        if core_transport::Keypair::public(&hopr_keys.packet_key).to_string().starts_with("0xff") {
+            warn!("This node uses an invalid packet key type and will not be able to become an effective relay node, please create a new identity!");
+        }
+
+        // Create the message inbox
+        let inbox: Arc<RwLock<hoprd_inbox::Inbox>> = Arc::new(RwLock::new(hoprd_inbox::inbox::MessageInbox::new_with_time(cfg.inbox.clone(), || {
+            platform::time::native::current_timestamp()
+        })));
+
+        let inbox_clone = inbox.clone();
+        let on_message = move |data: core_transport::ApplicationData| {
+            let now = Into::<DateTime<Utc>>::into(SystemTime::now()).to_rfc3339();
```

Comment:
`DateTime::<Utc>::from(...)` shorter?

Reply:
Indeed. This was a compiler recommended way at some point.